python-package/lightgbm/dask.py

```diff
@@ -333,15 +361,23 @@ def _predict_part(part, model, raw_score, pred_proba, pred_leaf, pred_contrib, *
     return result
 
 
-def _predict(model, data, raw_score=False, pred_proba=False, pred_leaf=False, pred_contrib=False,
-             dtype=np.float32, **kwargs):
+def _predict(
+    model: LGBMModel,
+    data: _DaskCollection,
+    raw_score: bool = False,
+    pred_proba: bool = False,
+    pred_leaf: bool = False,
+    pred_contrib: bool = False,
+    dtype: _PredictionDtype = np.float32,
+    **kwargs: Any
+) -> _DaskCollection:
```

Comment:
```suggestion
) -> dask_Array:
```

Based on https://github.com/microsoft/LightGBM/issues/3861#issuecomment-767970848, I realize now that this will always be a Dask array. That's the same behavior as the scikit-learn API.

```python
import numpy as np
import pandas as pd
from lightgbm import LGBMRegressor

reg = LGBMRegressor()

np.random.random(10)

num_features = 20
num_rows = 1000

X = pd.DataFrame({
    "col" + str(i): np.random.random(num_rows)
    for i in range(num_features)
})

y = np.random.random(num_rows)
reg.fit(X, y)

preds = reg.predict(X)

print(f"input type: {type(X)}, \npred type: {type(preds)}")
```